packages/jsii-java-runtime/project/src/main/java/software/amazon/jsii/JsiiEngine.java

```diff
@@ -578,6 +576,35 @@ public JsiiObjectRef createNewObject(final Object uninitializedNativeObject, fin
         return overrides.values();
     }
 
+    /**
+     * Crawls up the inheritance tree of {@code classToInspect} in order to determine the jsii-visible
+     * interfaces that are implemented, using a procedure akin to:
+     *
+     * <ol>
+     *  <li>If {@code classToInspect} is annotated with @Jsii and is an interface, return a singleton set with the FQN</li>
+     *  <li>If {@code classToInspect} is annotated with @Jsii and is NOT an interface, return an empty set</li>
+     *  <li>Collect all interfaces from {@classToInspect}'s interfaces</li>
+     * </ol>
```

Comment:
These types of comments are better off inline in the code close to where the logic that implements them lives. Otherwise, very good chance this will become stale at some point, and also harder to correlate to the actual code.